### ISSUE TITLE
Regularization in baselines

### ISSUE THREAD

**Issue by sophiagu** [hill-a/stable-baselines#981]
How can I do regularization (such as l1/l2, drop outs) in baselines?

**Comment by Miffyli** [comment]
Duplicate of #817 and #403.

See [docs on custom policies](https://stable-baselines.readthedocs.io/en/master/guide/custom_policy.html). You may need to modify loss functions for L1/L2/"weight decay" regularization, and that has to be done manually to the algorithm's code.

**Comment by sophiagu** [comment]
Thx Miffyli! Could you point me to the code location where I can modify to add regularization? I'm using the MlpLstmPolicy specifically.

**Comment by Miffyli** [comment]
Something [like this could do the trick](https://stackoverflow.com/a/38466108), which you then add to the loss. Losses are computed in the algorithm, e.g. [PPO2 here](https://github.com/hill-a/stable-baselines/blob/master/stable_baselines/ppo2/ppo2.py#L188). You may close the issue if there are no other bugs/issues to raise related to stable-baselines.

**Comment by huvar** [comment]
A good reference for this is the coinrun repository (https://github.com/openai/coinrun). Maybe it would be something easy and with little side-impacts to introduce in the master code.

https://github.com/openai/coinrun/blob/523704f3a203dcaad84caf96ea92799452dc902f/coinrun/ppo2.py#L105

**Comment by Miffyli** [comment]
One thing I forgot to mention: This is much easier in [PyTorch version of stable-baselines](https://github.com/DLR-RM/stable-baselines3), where you can add L2 regularization via the `weight_decay` parameter to optimizers. Note to self: We should probably expose this there.

**Comment by sophiagu** [comment]
Thx! One question: is stable-baselines moving to stable-baselines3 or is stable-baselines3 just a PyTorch version of this repo?

**Comment by Miffyli** [comment]
Our main focus is now on stable-baselines3 and we plan to mostly include bug fixes and small adjustments to this library.

**Comment by sophiagu** [comment]
Got it!
